File: src/models/reportModel.js
const fs = require('fs');
const path = require('path');

con​st reportsFile = path.join(__dirname, '../data/reports.json');

class ReportModel {
  static async getReports() {
    try {
      const rawData = fs.promises.readFile(reportsFile, 'utf8');
      const data = JSON.parse(rawData);
      return data;
    } catch (error) {
      console.error("Failed to load reports:", error);
      return [];
    }
  }
}

module.exports = ReportModel;

File: src/models/report_model.py
import json
import asyncio
from pathlib import Path
from typing import List, Any

reports_file = Path(__file__).resolve().parent / ".." / "data" / "reports.json"

class ReportModel:
    @staticmethod
    async def get_reports() -> List[Any]:
        try:
            def read_file():
                if not reports_file.exists():
                    return []
                return json.loads(reports_file.read_text(encoding="utf-8"))
            
            return await asyncio.to_thread(read_file)
        except Exception as error:
            print("Failed to load reports:", str(error))
            return []
